wait_for/__init__.py

```diff
@@ -89,6 +90,25 @@ def _get_failcondition_check(fail_condition):
         return partial(check_result_is_fail_condition, fail_condition)
 
 
+def _get_handled_exceptions(handle: Union[Type[Exception], Iterable[Type[Exception]]]) \
+        -> Iterable[Type[Exception]]:
+    if isinstance(handle, type) and issubclass(handle, Exception):
```

Comment:
This is a great pattern for handling this check overall, thanks for taking the time and effort to add this feature to `wait_for`. 

I think you can flatten this logic a bit, and the only risk will be something is passed to the function that isn't an `Exception` subclass will still get returned, which would cause a TypeError on the isinstance call inside the filter at L108.

```python
if isinstance(handle, Iterable):
    return iter(handle)
else:
    return iter(((Exception if isinstance(handle, bool) else handle),))
```

Maybe I'm missing something and there's more value to checking the subclass here, just something to consider.  I tried the above locally and still passed your new unit tests.

Reply:
I would like to give the user a possibility to pass the handle of Any type, not just of Union[bool, Type[Exception]] so, `isinstance(handle,` bool)` won't work in this case. Seem's I have to add more negative or edge test cases

Reply:
I've added more negative test cases to show what I mean